{
  "id": "GHSA-8x3m-m3x9-54fj",
  "summary": "JupyterHub OAuthenticator elevation of privilege",
  "details": "An issue was discovered in Project Jupyter JupyterHub OAuthenticator 0.6.x before 0.6.2 and 0.7.x before 0.7.3. When using JupyterHub with GitLab group whitelisting for access control, group membership was not checked correctly, allowing members not in the whitelisted groups to create accounts on the Hub. (Users were not allowed to access other users' accounts, but could create their own accounts on the Hub linked to their GitLab account. GitLab authentication not using gitlab_group_whitelist is unaffected. No other Authenticators are affected.)",
  "aliases": [
    "CVE-2018-7206"
  ],
  "modified": "2023-02-16T23:26:29.152300Z",
  "published": "2022-05-13T01:12:09Z",
  "database_specific": {
    "cwe_ids": [],
    "severity": "HIGH",
    "github_reviewed": true,
    "github_reviewed_at": "2022-09-12T16:17:03Z",
    "nvd_published_at": "2018-02-18T03:29:00Z"
  },
  "references": [
    {
      "type": "ADVISORY",
      "url": "https://nvd.nist.gov/vuln/detail/CVE-2018-7206"
    },
    {
      "type": "WEB",
      "url": "https://github.com/jupyterhub/oauthenticator/commit/1845c0e4b1bff3462c91c3108c85205acd3c75a2"
    },
    {
      "type": "WEB",
      "url": "https://blog.jupyter.org/security-fix-for-jupyterhub-gitlab-oauthenticator-7b14571d1f76"
    },
    {
      "type": "PACKAGE",
      "url": "https://github.com/jupyterhub/oauthenticator"
    },
    {
      "type": "WEB",
      "url": "https://github.com/jupyterhub/oauthenticator/blob/8499dc2/CHANGELOG.md#073---2018-02-16"
    }
  ],
  "affected": [
    {
      "package": {
        "name": "oauthenticator",
        "ecosystem": "PyPI",
        "purl": "pkg:pypi/oauthenticator"
      },
      "ranges": [
        {
          "type": "ECOSYSTEM",
          "events": [
            {
              "introduced": "0.6.0"
            },
            {
              "fixed": "0.6.2"
            }
          ]
        }
      ],
      "versions": [
        "0.6.0",
        "0.6.1"
      ],
      "database_specific": {
        "source": "https://github.com/github/advisory-database/blob/main/advisories/github-reviewed/2022/05/GHSA-8x3m-m3x9-54fj/GHSA-8x3m-m3x9-54fj.json"
      }
    },
    {
      "package": {
        "name": "oauthenticator",
        "ecosystem": "PyPI",
        "purl": "pkg:pypi/oauthenticator"
      },
      "ranges": [
        {
          "type": "ECOSYSTEM",
          "events": [
            {
              "introduced": "0.7.0"
            },
            {
              "fixed": "0.7.3"
            }
          ]
        }
      ],
      "versions": [
        "0.7.0",
        "0.7.1",
        "0.7.2"
      ],
      "database_specific": {
        "source": "https://github.com/github/advisory-database/blob/main/advisories/github-reviewed/2022/05/GHSA-8x3m-m3x9-54fj/GHSA-8x3m-m3x9-54fj.json"
      }
    }
  ],
  "schema_version": "1.3.0",
  "severity": [
    {
      "type": "CVSS_V3",
      "score": "CVSS:3.1/AV:N/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:H"
    }
  ]
}